server/resolvers/TestPlanRunOperations/createTestResultSkeleton.js

```diff
@@ -30,16 +30,26 @@ const createTestResultSkeleton = ({
                     id: scenarioResultId,
                     scenarioId: scenario.id,
                     output: null,
-                    assertionResults: test.assertions.map(assertion => {
-                        return {
+                    assertionResults: test.assertions
+                        // Filter out assertionResults which were marked with a 0-priority exception
+                        .filter(assertion => {
+                            return !assertion.assertionExceptions?.some(
+                                e =>
+                                    scenario.commands.find(
+                                        c =>
+                                            c.id === e.commandId &&
+                                            c.atOperatingMode === e.settings
+                                    ) && e.priority === 'EXCLUDE'
+                            );
+                        })
```

Comment:
The comment doesn't seem to acknowledge the first half of the filter expression, making me wonder whether I understand what's going on here. Would it be accurate to say, "Filter out assertionResults for the current scenario which were marked with a 0-priority exception"?

Stylistically, I think that the closure nesting is reaching a point where it interferes with readability. While the in-line documentation you've suggested helps, breaking this logic out into a standalone function would afford us more room to describe what's going on, reduce the cognitive load (by formalizing the inputs), and limit distraction for folks seeking to understand `createTestResultSkeleton` at a high level.

```js
/**
 * Determine whether a given assertion belongs to a given scenario and includes
 * at least one exception with a given priority.
 *
 * @param {string} priority
 * @param {Scenario} scenario
 * @param {Assertion} assertion
 */
const hasExceptionWithPriority = (priority, scenario, assertion) => {
  return assertion.assertionExceptions?.some(
      e =>
          scenario.commands.find(
              c =>
                  c.id === e.commandId &&
                  c.atOperatingMode === e.settings
          ) && e.priority === priority
  );
};
```

Then invoking like this:

    .filter(assertion => !hasExceptionWithPriority('EXCLUDE', scenario, assertion))

I'm only sharing something concrete as a means to demonstrate my point; I'm not too concerned with the specific factoring. For instance, a more purpose-built function would simplify the callsite even further (though it might be difficult to name it).

Reply:
+1 on this. I was able to take your suggestion and only apply minor changes, thanks! Done in https://github.com/w3c/aria-at-app/pull/863/commits/bba8ad4edb425d890effddf85dc9afa4aa535261